letsencrypt/client/client.py

```diff
@@ -389,18 +345,57 @@ def init_csr(privkey, names, cert_dir):
 
     return le_util.CSR(csr_filename, csr_der, "der")
 
+
 # This should be controlled by commandline parameters
-def determine_authenticator(config):
+def determine_authenticator(all_auths):
     """Returns a valid IAuthenticator.
 
-    :param config: Configuration.
-    :type config: :class:`letsencrypt.client.interfaces.IConfig`
+    :param list all_auths: Where each is a tuple of the form
+        ('description', 'IAuthenticator', *options..) where IAuthenticator is a
+        :class:`letsencrypt.client.interfaces.IAuthenticator` object or class
+        and options are the parameters used to initialize the authenticator.
+
+    :returns: Valid Authenticator object or None
 
     """
-    try:
-        return configurator.ApacheConfigurator(config)
-    except errors.LetsEncryptNoInstallationError:
-        logging.info("Unable to determine a way to authenticate the server")
+    # Available Authenticator objects
+    avail_auths = []
+    # Error messages for misconfigured authenticators
+    errs = {}
+
+    for pot_auth in all_auths:
+        try:
+            # I do not think this a great solution but haven't come up with
+            # anything better yet... other than constricting init functions for
+            # authenticators
```

Comment:
Separate sanity check function sounds great, especially in the light of the plugin system I'm proposing. You should be able to instantiate authenticator (even coming from outside our package) and ask it if the system is configured properly before allowing anyone to choose it in the CLI.

However, for now, this code is really confusing, and I'm sure you'll say the same after a week or so this is pulled in. ;)